src/comm/ScopedMpiInit.cc

```diff
@@ -56,8 +56,6 @@ ScopedMpiInit::ScopedMpiInit(int* argc, char*** argv)
                                   "ScopedMpiInit";
             break;
         }
-        default:
```

Comment:
Same comment as above.  Out of curiosity, what was the warning message?

Reply:
```
../src/comm/ScopedMpiInit.cc:59:9: warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]
```

Reply:
humm I wonder what they (the warning designed) expect for the case where the enum are used a bit fields:
```
enum {
  val0 = 0,
  val1 = 1,
  val2 = 1<<1,
  val3 = 1<<2,
// or
   val10, val11, va12,
   special_value = 1000,
}
```
```
switch (enumval) {
case val1 | val2: 
etc...
default: not_supported_yet();
};
or
switch (enumval) {
case val0: ... ; break;
case val0+special_value: ...; break;
};
```
(admittedly the first switch statement sounds very odd to use but the 2nd is a real use case .. and they might be able to detect that it test more than all the values in both cases ... )

Reply:
For the code:
```c++
enum class Color
{
    red,
    green,
    blue
};

const char* mix(Color c)
{
    switch (c)
    {
        case Color::red: return "red";
        case Color::green: return "green";
        case Color::blue: return "blue";
        default: return "nope";
    }
    return "unreachable";
}
```
I get:
```console
$ clang++ -Wall -Wextra -Weverything -pedantic -std=c++11 -Wno-c++98-compat -c test.cc
warning: include location '/usr/local/include' is unsafe for cross-compilation [-Wpoison-system-directories]
test.cc:15:9: warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]
        default: return "nope";
        ^
test.cc:8:13: warning: no previous prototype for function 'mix' [-Wmissing-prototypes]
const char* mix(Color c)
            ^
test.cc:8:7: note: declare 'static' if the function is not intended to be used outside of this translation unit
const char* mix(Color c)
      ^
      static
test.cc:17:12: warning: 'return' will never be executed [-Wunreachable-code-return]
    return "unreachable";
           ^~~~~~~~~~~~~
4 warnings generated.
```
Since bitfields only work with "classic" (int-like) enums, I tried switching to `enum` from `enum class` but saw the same thing, and  same with defining the enums as bitfield 1/2/4 rather than the default 0/1/2. Adding a  yellow enumeration for  red | green gives:
```
$ clang++ -Wall -Wextra -Weverything -pedantic -std=c++11 -Wno-c++98-compat -c test.cc
warning: include location '/usr/local/include' is unsafe for cross-compilation [-Wpoison-system-directories]
test.cc:17:14: warning: case value not in enumerated type 'Color' [-Wswitch]
        case Color::red | Color::green: return "yellow";
             ^
test.cc:18:9: warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]
        default: return "nope";
        ^
```
so it looks like clang really doesn't like enums being used as bits.